### ISSUE TITLE
optional chaining operator error

### ISSUE THREAD

**Issue by bosens-China** [babel/babel-eslint#811]
The question of "the"
Use the optional chaining operator prompt for an expression.

## Reappear
```js
Var a = {};
console.log(a?.b);
```
## config
```js
// babel.config.js
module.exports = {
  presets: [
    [
      "@babel/preset-env",
      {
        useBuiltIns: "usage",
        corejs: 3,
        modules: false
      }
    ]
  ],
  // 实验性语法
  plugins: ["@babel/plugin-proposal-class-properties", '@babel/plugin-proposal-optional-chaining']
};

```

```js
// .eslintrc.js
module.exports = {
  env: {
    browser: true,
    commonjs: true,
    es6: true,
    node: true
  },
  extends: "eslint:recommended",
  globals: {
    Atomics: "readonly",
    SharedArrayBuffer: "readonly"
  },
  parser: "babel-eslint",
  parserOptions: {
    ecmaVersion: 2018,
    sourceType: "module"
  },
  rules: {
    "no-console": process.env.NODE_ENV === "production" ? "error" : "off",
    "no-debugger": process.env.NODE_ENV === "production" ? "error" : "off",
 
  }
};
```

## Why is it so?
I tried to search for some questions. Some of them were downgraded to @ 8.2.1, which proved to be invalid. I want to know an exact answer

**Comment by nicolo-ribaudo** [comment]
What version of babel-eslint are you using? What unexpected behaviour are you getting?

Also, you should add `@babel/plugin-proposal-optional-chaining` to your Babel config.

**Comment by bosens-China** [comment]
> What version of babel-eslint are you using? What unexpected behaviour are you getting?
> 
> Also, you should add `@babel/plugin-proposal-optional-chaining` to your Babel config.

``` json
"eslint": "^6.7.2",
"babel-eslint": "^10.0.3",
```
## babel.config.js
``` js
module.exports = {
  presets: [
    [
      "@babel/preset-env",
      {
        useBuiltIns: "usage",
        corejs: 3,
        modules: false
      }
    ]
  ],
  // 实验性语法
  plugins: ["@babel/plugin-proposal-class-properties", '@babel/plugin-proposal-optional-chaining']
};
```

## Accident
``` js
Expression expected.
Expected ':'.
```

**Comment by kaicataldo** [comment]
Strange - 10.0.3 [should have this plugin enabled](https://github.com/babel/babel-eslint/blob/10.x/lib/parse.js#L42).

**Comment by existentialism** [comment]
Can't reproduce this locally... @bosens-China can you create a small repo that reproduces the problem?

**Comment by bosens-China** [comment]
> Can't reproduce this locally... @bosens-China can you create a small repo that reproduces the problem?

Of course, this is the new warehouse
[https://github.com/bosens-China/Test-file](https://github.com/bosens-China/Test-file)

**Comment by existentialism** [comment]
@bosens-China I cloned your repo, added `eslint` to `devDependencies`, and was able to run the `build` script without any errors?

**Comment by bosens-China** [comment]
> @bosens-China I cloned your repo, added `eslint` to `devDependencies` (it was missing), and was able to run `npm run build` without any errors?
> 
> ![image](https://user-images.githubusercontent.com/56288/70297271-fdece600-17b2-11ea-89b3-fcad2d49c5af.png)
> 
> and just to verify, I modified `src/main.js` with:
> 
> ```js
> var a = {};
> console.log(a?.b, c);
> ```
> 
> So it would force a lint error:
> 
> ![image](https://user-images.githubusercontent.com/56288/70297334-3f7d9100-17b3-11ea-859a-443f908ca981.png)

Thank you very much.
